libdnf/repo/repo_sack.cpp

```diff
@@ -352,17 +352,23 @@ void RepoSack::create_repos_from_file(const std::string & path) {
 }
 
 void RepoSack::create_repos_from_config_file() {
-    const auto & path = base->get_config().get_config_file_path_option().get_value();
+    std::filesystem::path conf_path{base->get_config().get_config_file_path_option().get_value()};
```

Comment:
Hmm, now there is the exactly same logic on two places - in `RepoSack::create_repos_from_config_file` and in `Base::load_config_from_file` except internally calling one different method `create_repos_from_file` vs `load_config_from_file` ... Maybe we could refactor it by using one method with logic and just passing the pointer to different internal one as an argument? Maybe it's a bit overkill, but on the other hand, I feel like when there will be some change in the logic in the future we might forget about the other place.

Reply:
Yeah, it would be nice not to duplicate the logic. Just pushed a commit that does it this way. It's a little awkward, but yes, this way the try/catch doesn't have to be repeated.

A good middle ground could be a `base.config_path()` that returns the correct path to the config file, possibly one inside the installroot, and then only the try/catch would be duplicated.